src/main/java/io/knotx/commons/http/request/AllowedHeadersFilter.java

```diff
@@ -35,18 +37,19 @@ public static AllowedHeadersFilter create(List<Pattern> patterns) {
     return new AllowedHeadersFilter(patterns);
   }
 
-  private static AllowedHeadersFilter create(Set<String> regexes) {
-    regexes = Objects.isNull(regexes) ? Collections.emptySet() : regexes;
-    List<Pattern> patterns = regexes.stream()
-        .map(regex -> Pattern.compile(regex, Pattern.CASE_INSENSITIVE))
+  private static AllowedHeadersFilter create(Set<String> expressions) {
+    expressions = Objects.isNull(expressions) ? Collections.emptySet() : expressions;
+    List<Pattern> patterns = expressions.stream()
+        .map(expression -> expression.replace(WILDCARD, WILDCARD_REGEX))
```

Comment:
Do we want to replace all `*` with `.*`?

Examples:
`*` -> `.*` (OK)
`Accept-*` -> `Accept-.*` (OK)
`Accept-.*` -> `Accept-..*` (INVALID)
`Accept-**` -> `Accept-.*.*` (?)

This change also requires some upgrade notes - projects can specify regexes now.

Reply:
I think we should just keep to one way of defining dynamic header names.
And I vote for simple regex, without any replacements.

Reply:
I reverted this changes.

In my opinion it be more useful to use regex instead of using wildcards as "*".